tiling.js

```diff
@@ -2515,18 +2515,17 @@ function ensuredX(meta_window, space) {
     let workArea = space.workArea();
     let min = workArea.x;
     let max = min + workArea.width;
-    if (meta_window.fullscreen) {
+    if (meta_window.fullscreen || meta_window.maximized_horizontally) {
         x = 0;
     } else if (index == 0 && x <= min) {
         // Always align the first window to the display's left edge
-        x = min;
+        x = min + prefs.horizontal_margin;
     } else if (index == space.length-1 && x + frame.width >= max) {
         // Always align the first window to the display's right edge
-        x = max - frame.width;
+        x = max - frame.width - prefs.horizontal_margin;
```

Comment:
Does it work to remove these special cases and let the code below handle it?

Reply:
Why yes, yes it does!  See #488 - thanks @smichel17.